aten/src/ATen/native/Sorting.cpp

```diff
@@ -216,6 +217,10 @@ std::tuple<Tensor&, Tensor&> kthvalue_out_impl_cpu(
   // FIXME: This seems bogus, I only do this because it was the old behaviour.
   //        The reductions are fine, as long as the axis being reduced along
   //        isn't of 0 elements (and the output has elements).
+  auto lap = at::get_overlap_status(self, values);
+  TORCH_CHECK(lap != at::MemOverlapStatus::PARTIAL &&
+                lap != at::MemOverlapStatus::FULL,
+                "Found overlap in output");
```

Comment:
I think this message should be a bit more clear. Maybe something like "output tensor's memory overlaps with input tensor"

Reply:
Done！